src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -0,0 +1,292 @@
+      { timelock, address }
+    );
+  }
+
+  res.send({
+    success: true,
+    data: proposal.partialTx.serialize(),
+    isComplete: proposal.partialTx.isComplete(),
+  });
+}
+
+async function getMySignatures(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const network = req.wallet.getNetworkObject();
+  const partialTx = req.body.partial_tx;
+  const proposal = PartialTxProposal.fromPartialTx(partialTx, network);
+
+  try {
+    storage.setStore(req.wallet.store);
+    await proposal.signData('123');
+    res.send({
+      success: true,
+      signatures: proposal.signatures.serialize(),
+      isComplete: proposal.signatures.isComplete(),
+    });
+  } catch (err) {
+    res.send({ success: false, error: err.message });
+  }
+}
+
+async function signTx(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const network = req.wallet.getNetworkObject();
+
+  const partialTx = req.body.partial_tx;
+  const signatures = req.body.signatures || [];
+
+  const proposal = PartialTxProposal.fromPartialTx(partialTx, network);
+
+  try {
+    await proposal.signData('123');
+    for (const signature of signatures) {
+      proposal.signatures.addSignatures(signature);
+    }
+    if (!proposal.isComplete()) {
+      throw new Error('Transaction is not complete');
+    }
+    const tx = proposal.prepareTx();
+    res.send({ success: true, txHex: tx.toHex() });
+  } catch (err) {
+    res.send({ success: false, error: err.message });
+  }
+}
+
+async function signAndPush(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const canStart = lock.lock(lockTypes.SEND_TX);
+  if (!canStart) {
+    res.send({ success: false, error: cantSendTxErrorMessage });
+    return;
+  }
+
+  const network = req.wallet.getNetworkObject();
+
+  const partialTx = req.body.partial_tx;
+  const signatures = req.body.signatures || [];
+
+  const proposal = PartialTxProposal.fromPartialTx(partialTx, network);
+
+  try {
+    storage.setStore(req.wallet.store);
+    await proposal.signData('123');
+    for (const signature of signatures) {
+      proposal.signatures.addSignatures(signature);
+    }
+    if (!proposal.isComplete()) {
+      throw new Error('Transaction is not complete');
+    }
+    const tx = proposal.prepareTx();
+
+    const sendTransaction = new SendTransaction({ transaction: tx, network });
+    const response = await sendTransaction.runFromMining();
+    res.send({ success: true, ...mapTxReturn(response) });
+  } catch (err) {
+    res.send({ success: false, error: err.message });
+  } finally {
+    lock.unlock(lockTypes.SEND_TX);
+  }
+}
+
+async function getInputData(req, res) {
```

Comment:
We don't have any tests for this API

Reply:
Created unit tests, the integration tests were already created.